pulser/sequence.py

```diff
@@ -1214,11 +1268,38 @@ def _target(
                     raise ValueError(
                         "All non-variable qubits must belong to the register."
                     )
-            return
 
-        elif not qubits_set.issubset(self._qids):
-            raise ValueError("All given qubits must belong to the register.")
+        return qubits_set
+
+    def _target(
+        self, qubits: Union[Iterable[QubitId], QubitId], channel: str
+    ) -> None:
+        qubits_set = self._precheck_target_qubits_set(qubits, channel)
+        if not self.is_parametrized():
+            self._perform_target_non_parametrized(qubits_set, channel)
+
+    @_store
+    def _target_index(
```

Comment:
So, the real reason we have `_target` and `_phase_shift` is that we need these to be called without being stored, not that they don't accept parametrized parameters. In this case, does it make sense to do the same or can you just store everything under `target_index()` and store that?

Reply:
Actually, the reason I did this was to drop the parametrization check from the stored function, since the deferred call happens when it's not parametrized anymore.

Reply:
I think what we want to avoid is that the register is mappable in a non-parametrized sequence, I think we should still allow it for non-parametrized sequences with normal registers (see my other comment)